ir/node.h

```diff
@@ -60,12 +68,19 @@ class INode : public Util::IHasSourceInfo, public IHasDbPrint, public ICastable
     virtual cstring node_type_name() const = 0;
     virtual void validate() const {}
     virtual const Annotation *getAnnotation(cstring) const { return nullptr; }
-    /// A checked version of INode::to. A BUG occurs if the cast fails.
-    ///
-    /// A similar effect can be achieved with `&as<T>()`, but this method
-    /// produces a message that is easier to debug.
+
+    // default checkedTo implementation for nodes: just fallback to generic ICastable method
+    template <typename T>
```

Comment:
Note that `static_type_name` is defined only on concrete classes, not on interfaces. So, without this change use of `checkedCastTo<T>` where `T` is interface would produce compile error. As a result, one was able to use `as<T>()` with `T` being interface, but not `checkedCastTo<T>()`.

Reply:
Hmm, maybe we should have `static_type_name` generated also for interfaces. That is a separate task though.

Reply:
Likely there was some rationale for this behavior. But maybe not. Though, I tend to agree that having a static type name everywhere would be be better.

Reply:
>Likely there was some rationale for this behavior. 

I added the ICastable class at some point because I noticed we repeated this pattern a lot. At the time `node_type_name` and `static_type_name` were simply not required. There is no reason not to add them, but it will likely take some effort across the entire compiler.

Reply:
@fruffy Sorry, I do not follow. What are you suggesting? Indeed we can get rid of this implementation, however we'll end with less friendly error message. Some tests checks this btw.

Reply:
Just clarifying on 
>Likely there was some rationale for this behavior.

There was not a concrete reason why things were done this way. No need to get rid of the implementation.

Reply:
Yes, there were a few nodes that implemented their own `checkedTo` and `to` but did not add the other functions. This class just unified things, the mangled typename was, well, "good enough".